src/Entity/Document.php

```diff
@@ -393,7 +393,8 @@ public function hasParentFolder(): bool
     public function move(?Dossier $parentFolder): void
     {
         $this->setDossier($parentFolder);
-        if ($parentFolder) {
+
+        if (true === $parentFolder?->getBPrive()) {
             $parentFolder->addDocument($this);
             $this->setBPrive($parentFolder->getBPrive());
```

Comment:
Considering the if condition, i don't think we need to get the value
`$this->setBPrive(true); `